src/mixins/itext_behavior.mixin.js

```diff
@@ -341,10 +341,28 @@
       this._setEditingProps();
 
       this._tick();
-      this.canvas && this.canvas.renderAll();
-
       this.fire('editing:entered');
-      this.canvas && this.canvas.fire('text:editing:entered', { target: this });
+      if (this.canvas) {
```

Comment:
Yes, I thought about early return at first, coz it's simple and makes it easier to read afterwards. But then I saw you removed `initMousemoveHandler` and I realized that having all that code in a separate function is even better.